### PR TITLE
Add bold and italic format guidance

### PR DESCRIPTION
**Pull request by josmperez**
Add guidance around using bold and italic formatting for emphasis. Hmmm... didnn't see any really obvious place for this addition in the existing structure, so I put it in the least wrong (?) place.

Implements https://docs.google.com/document/d/1uCjtHEy0lpZggkpgsLDjWwxV9K6h7vcsG2dy4TOS6EA/edit

### CODE REVIEW

File: docs/sources/write/style-guide/style-conventions/index.md [hunk 1]
```diff
@@ -244,3 +244,19 @@ When you write in active voice, you identify the subject of the sentence and the
...
+
+### Bold
+
+Use bold formatting, \*\* in Markdown, for UI elements. Don't use bold to draw attention to a word or phrase.
+
+### Italic
+
+Use italics formatting, \_ in Markdown, to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.
+
+**Recommended:** The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.
```
Comment (jdbaldry) [inline docs/sources/write/style-guide/style-conventions/index.md:260]: Do we need to mention this use case for bold emphasis?

**Comment by jdbaldry** [comment]
Looks like you were caught out by https://github.com/grafana/writers-toolkit/issues/307.

**Comment by jdbaldry** [comment]
I rewrote the history on this branch to resolve the prettier error and clean up the commits.

File: docs/sources/write/style-guide/style-conventions/index.md (hunk 1 above)
Comment (josmperez) [reply docs/sources/write/style-guide/style-conventions/index.md:260]: no, shorter is better

**Comment by JStickler** [comment]
I think it would also make sense to link to /docs/sources/write/markdown-guide, so that if someone is looking for the markup to create bold and italics, they're also directed to the guidelines for using it.

File: docs/sources/write/style-guide/style-conventions/index.md (hunk 1 above)
Comment (jdbaldry) [reply docs/sources/write/style-guide/style-conventions/index.md:260]: I personally found it odd that we immediately used bold emphasis immediately after we said that we shouldn't use bold to draw attention to a word or phrase.

File: docs/sources/write/style-guide/style-conventions/index.md [hunk 2]
```diff
@@ -244,3 +244,17 @@ When you write in active voice, you identify the subject of the sentence and the
...
 The output is the same in both cases.
 However, the first is easier to review and edit and is less subject to the screen and text editor settings of each contributor.
+
+## Text formatting
+
+It's a good idea to take a consistent approach to bold, italic, and other text formats.
+
+### Bold
+
+Use bold formatting, \*\* in Markdown, for UI elements. Don't use bold to draw attention to a word or phrase.
```
Comment (JStickler) [inline docs/sources/write/style-guide/style-conventions/index.md:254]:
```suggestion
Use bold formatting (\*\* in Markdown), for UI elements. Don't use bold to draw attention to a word or phrase.
```

File: docs/sources/write/style-guide/style-conventions/index.md [hunk 3]
```diff
@@ -244,3 +244,17 @@ When you write in active voice, you identify the subject of the sentence and the
...
+
+### Italic
+
+Use italics formatting, \_ in Markdown, to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.
+
+**Recommended:** The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.
```
Comment (jdbaldry) [inline docs/sources/write/style-guide/style-conventions/index.md:260]: This suggestion avoids the confusion of using bold in a different way to the docs.

```suggestion
For example:

> The Prometheus data model is arranged into _metrics_ that consist of a _timestamp_ and a _sample_.
```

File: docs/sources/write/style-guide/style-conventions/index.md [hunk 4]
```diff
@@ -244,3 +244,17 @@ When you write in active voice, you identify the subject of the sentence and the
...
+
+It's a good idea to take a consistent approach to bold, italic, and other text formats.
+
+### Bold
+
+Use bold formatting, \*\* in Markdown, for UI elements. Don't use bold to draw attention to a word or phrase.
+
+### Italic
+
+Use italics formatting, \_ in Markdown, to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.
```
Comment (JStickler) [inline docs/sources/write/style-guide/style-conventions/index.md:258]:
```suggestion
Use italics formatting (\_ in Markdown), to emphasize a specific word or phrase. This is particularly useful when defining a term for the first time.
```

**Review by imatwawana — APPROVED** [review]

**Pull request merged**